gnosis/test/verification.k

```diff
@@ -117,14 +119,13 @@ module VERIFICATION
        andBool MASKWIDTH0 ==Int #sizeWordStack(WS1)
 
     // Arithmetic
-    rule I1 +Int (I2 +Int I3) => I2 +Int (I1 +Int I3) when #isConcrete(I1) andBool #isConcrete(I3)
-    rule I1 +Int (I2 +Int I3) => I3 +Int (I1 +Int I2) when #isConcrete(I1) andBool #isConcrete(I2)
-
-    rule (I1 +Int I2) +Int (I3 -Int I2) => I1 +Int I3
-    rule (I1 +Int I2) -Int (I1 +Int I3) => I2 -Int I3
+    rule (A +Int I2) +Int (I3 -Int A) => I2 +Int I3
+    rule (A +Int I2) -Int (A +Int I3) => I2 -Int I3
+		rule (A +Int I2) -Int (I3 +Int A) => I2 -Int I3
```

Comment:
@denis-bogdanas could you explain what happens here?

Reply:
Trying to separate symbolic terms of +/- from concrete terms, so that concrete terms all add up at the same side of +Int and no matter which path is taken by some other functions, final expression always looks the same.

Principles are explained in comments: move symbolic terms always to the left, convert -Int A into +Int (0 -Int A) to have always +Int, except when LHS is 0 or another symbolic term. This drastically reduces the number of particular cases elsewhere, like the +Int simplification expressions in verification.k

Reply:
That's not what i'm asking.  I started a new thread below.